internal/versions/generateHashes.go

```diff
@@ -0,0 +1,105 @@
+//go:build ignore
+
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package main
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/printer"
+	"go/token"
+	"log"
+	"net/http"
+	"os"
+
+	"golang.org/x/tools/go/ast/astutil"
+)
+
+func mustGetHash(url string) string {
+	buf := bytes.Buffer{}
+
+	// remove quotes around url
+	url = url[1 : len(url)-1]
+
+	// Get the data
+	resp, err := http.Get(url)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	// Check server response
+	if resp.StatusCode != http.StatusOK {
+		panic("bad status: " + resp.Status)
+	}
+
+	// Writer the body to buffer
+	if _, err = buf.ReadFrom(resp.Body); err != nil {
+		panic(err)
+	}
+
+	// Generate SHA256 hash of the file
+	h := sha256.New()
```

Comment:
Have we considered using the hashes provided by the publishers of the respective project? Or, why did we decide to calculate the hashes ourselves?

For `crictl` e.g. there is:
* release: https://github.com/kubernetes-sigs/cri-tools/releases/download/v1.25.0/crictl-v1.25.0-linux-amd64.tar.gz
* hash256: https://github.com/kubernetes-sigs/cri-tools/releases/download/v1.25.0/crictl-v1.25.0-linux-amd64.tar.gz.sha256
* hash512: https://github.com/kubernetes-sigs/cri-tools/releases/download/v1.25.0/crictl-v1.25.0-linux-amd64.tar.gz.sha512

Reply:
Alternatively, we could also do both and compare the result.

Reply:
Opted for your 2nd suggestion. Taking both the upstream hash and hashing the file ourselves gives us a compile/generate time check if the hashes match and therefore if our Constellation setup will succeed.